services/src/floodlight-modules/src/main/java/org/openkilda/floodlight/switchmanager/SwitchManager.java

```diff
@@ -712,23 +707,38 @@ public void installVerificationRule(final DatapathId dpid, final boolean isBroad
         // Don't install the unicast for OpenFlow 1.2 doesn't work properly
         if (!isBroadcast) {
             if (ofFactory.getVersion().compareTo(OF_12) > 0) {
-                logger.debug("installing unicast verification match for {}", dpid);
+                logger.debug("Installing unicast verification match for {}", dpid);
             } else {
-                logger.debug("not installing unicast verification match for {}", dpid);
+                logger.debug("Not installing unicast verification match for {}", dpid);
                 return;
             }
         }
 
-        logger.debug("installing verification rule for {}", dpid);
+        logger.debug("Installing verification rule for {}", dpid);
 
-        Match match = matchVerification(sw, isBroadcast);
-        ArrayList<OFAction> actionList = new ArrayList<>(2);
+        long cookie = isBroadcast ? VERIFICATION_BROADCAST_RULE_COOKIE : VERIFICATION_UNICAST_RULE_COOKIE;
+        long meterId = cookie & PACKET_IN_RULES_METERS_MASK;
+        long meterRate = isBroadcast ? config.getBroadcastRateLimit() : config.getUnicastRateLimit();
+        if (!meterExists(sw, meterId, meterRate)) {
+            installMeterForDefaultRule(sw, meterId, meterRate);
+        } else {
+            logger.debug("Meter {} won't be installed on {} switch. It already exists", meterId, dpid);
+        }
+
+        ArrayList<OFAction> actionList = new ArrayList<>(3);
         actionList.add(actionSendToController(sw));
         actionList.add(actionSetDstMac(sw, dpidToMac(sw)));
+
+        OFInstructionMeter meter = null;
+        //FIXME: we should install meters on all switches with OF > 1.2
+        if (isSwitchSupportsPktpsFlag(sw)) {
```

Comment:
Why we have this condition here?
We should catch exception `UnsupportedSwitchOperationException` from `installMeterForDefaultRule` and make appropriate decisions here and not reinvent internal logic of `installMeterForDefaultRule`.